charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml

```diff
@@ -49,6 +49,17 @@ configMap:
   # -- Annotations to add to the SPIFFE OIDC Discovery Provider ConfigMap
   annotations: {}
 
+testSecurityContext:
+  allowPrivilegeEscalation: false
+  runAsNonRoot: true
+  runAsUser: 10001
+  runAsGroup: 10001
+  seccompProfile:
+    type: RuntimeDefault
+  capabilities:
+    drop:
+    - ALL
```

Comment:
Can these be moved to examples/production/values.yaml for now to be consistent with all the other similar settings?

Reply:
The purpose is to be able to run these post install in real deployments, so they need to be defined here

Reply:
It should work without them?

Just trying to be consistent with whats already in the tree. All this kind of stuff is under the production example config.

Reply:
all of those overrides are required if they follow install instructions by putting pod restrictions on the namespace.

Reply:
Whatever is in production doesn't apply here, testSecurityContext is new. This PR is for me to run tests in my production environment, not just to add more tests to github actions. I want to verify Spire is running on our production environment. I would not consider this PR complete without the values required to make the tests functional. `helm test` should be portable, and these overrides are required because of the spire install instructions: https://github.com/spiffe/helm-charts/tree/main/examples/production#recommended-production-setup

Reply:
I still don't follow... how does helm test fail if you don't use -f examples/production/values.yaml without the added test config. If your using -f examples/production/values.yaml, how does it fail if these settings were with the rest in that file?

Reply:
The use case is in my production environment. 

1. Install chart via cicd
2. CICD installs chart with test information in manifest
3. log into production k8s cluster
4. `helm test -n {namespace} spire`

There's no config provided in #4, I would not expect people to run `helm test -n {ns} spire -f spiffe/helm-charts@{exact version of chart installed in production}`. The test needs to be portable and utilize settings provided when the chart itself was installed in the cluster.

Reply:
Right. how does that fail without the stuff in https://github.com/spiffe/helm-charts/pull/423/files#diff-7be84ce176a12ba3b77f98055916d01a5c97e1688a3a32e361495839b5ef69caR53-R61?

By default it should work I think. Unless your doing something to configure the chart / namespace like whats in examples/production and you'd need all of those values to function in that config, not just the testing ones?

Reply:
We should make the tests work with the recommended install steps. If we're going to tell people to restrict pods, the tests should work in a restricted environment https://github.com/spiffe/helm-charts/tree/main/examples/production#recommended-production-setup

Reply:
The recommended production setup includes examples/production/values.yaml. You gotta use all those settings or it will break. Having the values here just gives a false sense that it will work without it, which it wont. the csi driver will break, the spire-agent will break, etc. Moving the settings there makes sure it either all works together or not at all.

In another issue/pr, we can look at merging everything in examples/production/values.yaml into the chart proper. but thats a much bigger issue/pr then belongs here.